src/status_im/transport/inbox.cljs

```diff
@@ -357,7 +357,10 @@
 (handlers/register-handler-fx
  :inbox/check-fetching
  (fn [{:keys [db now] :as cofx} [_ last-request chat-id]]
-   (when (:inbox/fetching? db)
+   (when (and (:inbox/fetching? db)
+              ;; if chat was removed before messages were fetched no need
+              ;; to proceed with further actions
+              (or (not chat-id) (contains? (:transport/chats db) chat-id)))
```

Comment:
this event can also be called for fetching of all chats, when chat-id is `nil`, (contains? m nil) is false